contracts/colony/ColonyStorage.sol

```diff
@@ -107,6 +107,9 @@ contract ColonyStorage is ColonyDataTypes, ColonyNetworkDataTypes, DSMath, Commo
   uint256 constant METATRANSACTION_NONCES_SLOT = 35;
   mapping(address => uint256) metatransactionNonces; // Storage slot 35
 
+  uint256 rootLocalSkill; // Storage slot 36
+  mapping (uint256 => bool) localSkills; // Storage slot 37
```

Comment:
After thinking about it, I'm leaning towards leaving it unprotected -- I agree that there's no external damage that could be done (all this would allow is colonies setting a domain skill, or another colony's skill) on their own data. I could imagine some quirky use-case where several colonies decide to use the same set of local skill ids, and so they use recovery mode to allow them in their own colonies. As this rep is still colony-specific, it doesn't matter too much. I agree that the `skillExists` check is important in making sure that this can only affect skills which "already exist". You also still can't set deprecated skills.

Reply:
Yeah, I guess if neither of us can do worse than self-sabotage, it's okay. I'll keep throwing myself against the barricade for a while longer, I think.

Reply:
It's marginal, but the _existence_ of those other skills does cross colony boundaries. I could spam a colony's skill tree, for example, rendering whatever UI exists in the app potentially unusable.

Reply:
How does that work? This mapping only allows "outside" skills to be erroneously used in this colony. There are many skills which do not show up in a colony's skill tree (other colony domain skills, for instance).

Reply:
D'oh, you're right, because this local skill tree is only one layer deep, so `addLocalSkill` doesn't take a parameter like the colonyNetwork one.

Such a thing is still possible currently by editing the skillId associated with a domain though, I think?

Reply:
What would that do? Ultimately any rep earned will be tagged with the colony as well as the skillId, so even if a colony manipulates it's internal accounting to give it's members reputation in odd skills, all that rep will still be colony-specific.

Reply:
Oh, you're saying a colony could:

- Use recovery mode to edit a domain struct to contain a skillId from another colony
- Then create new child domains from that domain, which would in fact create new skills in the other colony's skill tree

That's interesting, since the other colony wouldn't be aware of those skills (there would be no corresponding domain or local skill record). But they would exist in the skill tree... I wonder how we could guard against this.

Reply:
Yeah, exactly. You're right there'd be no local skill record in the corresponding colony though, so I guess if the frontend checks against the local skill record for each skill it finds in the tree, that limits the griefing that can be done. Being in the skill tree means people will have 0 reputation in these skills once they earned a negative reputation further up the tree, but again, not a huge issue, as it doesn't change how reputation would be awarded/penalised.

I can't see a way of guarding against this (for existing colonies).